Server.py

```diff
@@ -1,47 +1,144 @@
 import socket
+import threading
 from Data import parse_database_file
 
-# Define the path to your database file
-database_file = "Pink_Floyd_DB.txt"
+# Constants
+DATABASE_FILE = "Pink_Floyd_DB.txt"
+SERVER_HOST = 'localhost'
+SERVER_PORT = 12345
+BUFFER_SIZE = 1024
+WELCOME_MESSAGE = b"Welcome to the Pink Floyd server. Please choose a command.\n"
+GOODBYE_MESSAGE = b"Goodbye!"
+INVALID_COMMAND_MESSAGE = b"Invalid command. Please try again.\n"
+
+# Load albums
+albums = parse_database_file(DATABASE_FILE)
 
 
-# Function to handle the command for listing albums
 def handle_list_albums():
-    albums_data = parse_database_file()
-    return list(albums_data.keys())
+    return [album['name'] for album in albums]
 
 
-# Example of server setup and main execution loop
-def main():
-    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    server_socket.bind(('localhost', 12345))
-    server_socket.listen(5)
-    print("Server is listening on port 12345...")
+def handle_list_songs_in_album(album_name):
+    for album in albums:
+        if album['name'].lower() == album_name.lower():
+            return [song['name'] for song in album['songs']]
+    return None
+
+
+def get_song_detail(song_name, detail):
+    for album in albums:
+        for song in album['songs']:
+            if song['name'].lower() == song_name.lower():
+                return song.get(detail, "Detail not found.")
+    return "Song not found."
+
+
+def handle_album_of_song(song_name):
+    for album in albums:
+        for song in album['songs']:
+            if song['name'].lower() == song_name.lower():
+                return album['name']
+    return None
+
+
+def handle_search_song(query, key):
+    results = []
+    for album in albums:
+        for song in album['songs']:
+            if query.lower() in song[key].lower():
+                results.append(song['name'])
+    return results
+
+
+def send_response(client_socket, response):
+    client_socket.send(response.encode())
+
+
+def receive_input(client_socket, prompt=None):
+    if prompt:
+        client_socket.send(prompt.encode())
+    return client_socket.recv(BUFFER_SIZE).decode().strip()
 
-    try:
-        while True:
-            client_socket, client_address = server_socket.accept()
-            print(f"Connection from {client_address} has been established.")
 
-            client_socket.send(b"Welcome to the Pink Floyd server. Please choose a command.\n")
+def handle_client(client_socket):
+    client_socket.send(WELCOME_MESSAGE)
 
-            while True:
-                command = client_socket.recv(1024).decode().strip()
+    while True:
+        command = receive_input(client_socket)
 
-                if command == "1":
-                    albums_list = handle_list_albums()
-                    response = "\n".join(albums_list)
-                    client_socket.send(response.encode())
+        if command == "1":
```

Comment:
this if elif elif.... has alot of duplicate code.
also, its better practice to use a dictionary that point to functions instead.
use a dict and resolve duplicate code